File: sample/client/ts/engine/notice_cli.ts
import * as engine from "./engine";
import { encode, decode } from "./@msgpack/msgpack";
import * as common from "./common_cli";
// this enum code is codegen by geese codegen for ts

// this struct code is codegen by geese codegen for ts
// this module code is codegen by geese codegen for typescript
export class notice_module {
    public on_notice:((session, string) => void)[] = []
    public constructor() {
        engine.app.instance.register_global_method("notice", this.notice)
    }

    public notice(hub_name:string, bin:Uint8Array) {
        let inArray = decode(bin) as any;
        let _msg = inArray[0];
        let s = new engine.session(hub_name)
        for (let fn of this.on_notice) {
            fn(s, _msg);
        }
    }

}

File: sample/client/py/engine/notice_cli.py
from threading import Timer
from collections.abc import Callable
from enum import Enum
from .engine import *
from .engine.msgpack import *
from .common_cli import *

# this enum code is codegen by geese codegen for python

#this struct code is codegen by geese codegen for python
#this module code is codegen by geese codegen for python
class notice_module(object):
    def __init__(self):
        pass

        self.on_notice:list[Callable[[session, str], None]] = []
        app().register_global_method("notice", self.notice)

    def notice(self, hub_name:str, bin:bytes):
        inArray = loads(bin)
        _msg = inArray[0]
        s = session(hub_name)
        for fn in self.on_notice:
            fn(s, _msg)




